api/lib/src/lib.rs

```diff
@@ -345,6 +345,18 @@ impl fmt::Display for WithdrawFeesDetail {
 	}
 }
 
+macro_rules! extract_event {
+    ($events:expr, $event_variant:path, $pattern:tt, $result:expr) => {
+        if let Some(state_chain_runtime::RuntimeEvent::Swapping($event_variant $pattern)) = $events.iter().find(|event| {
+            matches!(event, state_chain_runtime::RuntimeEvent::Swapping($event_variant { .. }))
```

Comment:
We can probably make this work for any pallet with a bit of into/try_into gymnastics:

```suggestion
        if let Some($event_variant $pattern) = $events.iter()
        	.filter_map(|runtime_event| runtime_event.try_into().ok())
        	.find(|pallet_event| { // <-- might need explicit type annotation
            	matches!(pallet_event, $event_variant { .. })
```

Might have to split the pallet event type and variant.